profile/src/main/java/rdx/works/profile/data/repository/HostInfoRepository.kt

```diff
@@ -1,137 +1,30 @@
 package rdx.works.profile.data.repository
 
-import android.content.Context
-import android.os.Build
-import android.provider.Settings
-import androidx.annotation.VisibleForTesting
-import androidx.core.content.edit
-import com.radixdlt.sargon.DeviceInfoDescription
 import com.radixdlt.sargon.HostId
 import com.radixdlt.sargon.HostInfo
-import com.radixdlt.sargon.HostOs
-import com.radixdlt.sargon.Timestamp
-import com.radixdlt.sargon.Uuid
-import com.radixdlt.sargon.extensions.android
-import dagger.hilt.android.qualifiers.ApplicationContext
-import kotlinx.serialization.Serializable
-import kotlinx.serialization.encodeToString
-import kotlinx.serialization.json.Json
-import rdx.works.core.UUIDGenerator
-import rdx.works.core.serializers.TimestampSerializer
-import rdx.works.core.serializers.UuidSerializer
-import timber.log.Timber
-import java.util.Locale
+import com.radixdlt.sargon.os.SargonOsManager
 import javax.inject.Inject
 
 interface HostInfoRepository {
 
-    fun getHostId(): HostId
+    suspend fun getHostId(): Result<HostId>
 
-    fun getHostInfo(): HostInfo
+    suspend fun getHostInfo(): Result<HostInfo>
 }
 
 class HostInfoRepositoryImpl @Inject constructor(
-    @ApplicationContext private val context: Context
+    private val sargonOsManager: SargonOsManager
 ) : HostInfoRepository {
 
-    override fun getHostId(): HostId {
-        val hostId = getStoredHostId(context)
+    override suspend fun getHostId(): Result<HostId> {
+        val sargonOs = sargonOsManager.sargonOs
 
-        return hostId ?: generateHostId(context)
+        return runCatching { sargonOs.resolveHostId() }
```

Comment:
@sergiupuhalschi-rdx HostInfo should not fail, and you are right there is no need to propagate the error to the user. On the other hand HostId can only fail if sargon fails to save the newly generated ID (take a look [here](https://github.com/radixdlt/sargon/blob/752616f8bf26fdc071c78c835f5e3976ddc465aa/crates/sargon/src/system/sargon_os/sargon_os.rs#L233)). For this in order to throw this [kotlin driver](https://github.com/radixdlt/sargon/blob/752616f8bf26fdc071c78c835f5e3976ddc465aa/jvm/sargon-android/src/main/java/com/radixdlt/sargon/os/storage/key/HostIdKeyMapping.kt#L38) needs to fail. But in general it shouldn't. What is your opinion on this?

Reply:
By looking at the code, all the reasons it would fail would fall into developer error category, explored these: `newHostIdFromJsonBytes`, `Json.encodeToString`, `deviceStorage.write(..)`. It looks like it's fine to treat it the same as the `HostInfo`. But you definitely have more context and know better.

Reply:
You are right I have reverted all of such functions to "not throw"